sec_edgar_downloader/rate_limiter.py

```diff
@@ -0,0 +1,32 @@
+import typing
+
+from datetime import datetime
+
+PWN = typing.TypeVar('PWN')
+CENTRAL_STATE = {}
+
+class RateLimiter:
+    def __init__(self: PWN, session_name: str) -> None:
+        self._name = session_name
```

Comment:
What is your rationale for the session name? I think we should use one central rate limiter that can be invoked in multiple places, and we can mock the `CENTRAL_STATE` in tests.